.github/workflows/integration-test.yaml

```diff
@@ -45,9 +45,13 @@ on:
 jobs:
   test:
     timeout-minutes: 120
-    name: Test ${{ inputs.runs-on }} for Mercurial ${{ inputs.hg-version }} on ${{ inputs.environment }}
+    name: ${{ matrix.test }} tests for HG ${{ inputs.hg-version }}
     permissions:
       checks: write
+    strategy:
```

Comment:
oof, this get's ugly pretty fast. Why don't we make 2 jobs? Right now we have the test job, we should split them out so we just have 2 with different steps, instead of this where we only run steps based on the matrix.

Reply:
Because I missed the line "A workflow run is made up of one or more jobs, **which run in parallel by default**" (emphasis mine) in https://docs.github.com/en/actions/using-jobs/using-jobs-in-a-workflow. You're quite right, this is ugly, but I thought it was the only way to do it. I'll revert this and take the much more sensible approach of using two jobs now that I realize it's possible.